projects/vdk-plugins/vdk-test-utils/src/vdk/plugin/test_utils/ingest_util_plugins.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import sys
+from dataclasses import dataclass
+from typing import List
+from typing import Optional
+
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.api.plugin.plugin_input import IIngesterPlugin
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+
+
+log = logging.getLogger(__name__)
+
+
+class ConvertPayloadValuesToString(IIngesterPlugin):
+    def pre_ingest_process(
+        self, payload: List[dict], metadata: IIngesterPlugin.IngestionMetadata = None
+    ) -> List[dict]:
+        return [{k: str(v) for (k, v) in i.items()} for i in payload]
+
+    @hookimpl(tryfirst=True)
+    def initialize_job(self, context: JobContext) -> None:
+        log.info("Initialize data job with ConvertPayloadValuesToString Plugin.")
+
+        context.ingester.add_ingester_factory_method("convert-to-string", lambda: self)
+
+
+class AddPayloadSizeAsColumn(IIngesterPlugin):
+    @dataclass
+    class Payload:
+        payload: List[dict]
+        destination_table: Optional[str]
+        target: Optional[str]
+        collection_id: Optional[str]
+
+    def __init__(self):
+        self.payloads: List[AddPayloadSizeAsColumn.Payload] = []
+
+    # Do payload pre-processing
+    def pre_ingest_process(
+        self, payload: List[dict], metadata: IIngesterPlugin.IngestionMetadata = None
+    ) -> List[dict]:
+        processed_payloads = []
+        for i in payload:
+            payload_size = sys.getsizeof(str(i))
+            i["payload_size"] = payload_size
+            processed_payloads.append(i)
+
+        return processed_payloads
+
+    # Ingest pre-processed payload
+    def ingest_payload(
+        self,
+        payload: List[dict],
+        destination_table: Optional[str],
+        target: Optional[str] = None,
+        collection_id: Optional[str] = None,
+    ):
+        self.payloads.append(
+            AddPayloadSizeAsColumn.Payload(
+                payload, destination_table, target, collection_id
+            )
+        )
+
+    # Process ingestion metadata
+    def post_ingest_process(
+        self,
+        payload: Optional[List[dict]] = None,
+        ingestion_metadata: Optional[IIngesterPlugin.IngestionMetadata] = None,
+        exceptions: Optional[List] = None,
+    ) -> None:
+        pass
```

Comment:
for a sample pre- and post- IngestionMetadata, would suggest populating ingestion submission duration -> start and end time

Reply:
Ok, but the start and end time would vary between test runs, which would make it hard to properly test. Or do you mean that some static values for start and end time are to be added?

Reply:
yeah, could either seed them dynamically or statically; then verify metadata propagation to post processing

Reply:
Done, thanks